File: src/main/kotlin/y2017/Day05.kt
package y2017

import utils.Day
import utils.readers.Reader
import utils.Helpers.jump
import utils.collections.mut
import utils.readers.asInts

class Day05 : Day<Int> {

    private val input = file.asInts()

    override fun runAll() = super.run({partOne(input)}, {partTwo(input)})

    private fun partOne(input: List<Int>) = jump(input.mut(), { 1 })

    private fun partTwo(input: List<Int>) = jump(input.mut(), { if (it >= 3) -1 else 1 })
}

File: rust/src/lib/years/y2021/day05.rs
use crate::lib::utils::tday::TDay;

pub struct Day05 {
    pub file: String,
    pub desc: String,
}

impl TDay for Day05 {
    fn desc(&self) -> String {
        (&self.desc).to_string()
    }

    fn part_one(&self) -> String {
        format!("{}", 2)
    }

    fn part_two(&self) -> String {
        format!("{}", 2)
    }
}